src/Microsoft.Windows.CsWin32/templates/ComHelpers.cs

```diff
@@ -10,4 +10,19 @@
 		@object = ComWrappers.ComInterfaceDispatch.GetInstance<TInterface>((ComWrappers.ComInterfaceDispatch*)@this);
 		return @object is null ? COR_E_OBJECTDISPOSED : S_OK;
 	}
+
+#if canUseInterfaceStaticMembers
+	internal static void PopulateIUnknown<TComInterface>(System.Com.IUnknown.Vtbl* vtable)
+		where TComInterface : unmanaged
+	{
+		PopulateIUnknownImpl<TComInterface>(vtable);
+		if (vtable->QueryInterface_1 is null)
+		{
+			throw new NotImplementedException("v-tables cannot be accessed unless the Windows.Win32.ComHelpers.PopulateIUnknownImpl partial method is implemented.");
+		}
```

Comment:
This is the extent of our runtime check to ensure that if the user wants the vtable that they have implemented the partial method.

Reply:
That should be good, imo.